Code/GraphMol/FileParsers/XYZFileWriter.cpp

```diff
@@ -0,0 +1,57 @@
+// $Id$
+//
+//  Copyright (C) 2019 Eisuke Kawashima
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include "FileParsers.h"
+#include <fstream>
+#include <sstream>
+#include <boost/format.hpp>
+#include <RDGeneral/BadFileException.h>
+
+namespace RDKit {
+
+std::string MolToXYZBlock(const ROMol &mol, int confId) {
+  if (!mol.getNumConformers()) {
+    BOOST_LOG(rdErrorLog)
+        << "Cannot write molecules with no conformers to XYZ block\n";
+    return "";
+  }
+
+  const auto& conf = mol.getConformer(confId);
+  const unsigned int nAtoms = mol.getNumAtoms();
+
+  std::stringstream ss;
+  ss << nAtoms << '\n';
+
+  std::string name;
+  if (mol.getPropIfPresent(common_properties::_Name, name)) {
+    ss << name.substr(0, name.find_first_of('\n'));
+  }
+  ss << '\n';
+
+  for (unsigned int i = 0; i < nAtoms; i++) {
+    const auto& symbol = mol.getAtomWithIdx(i)->getSymbol();
+    const auto& pos = conf.getAtomPos(i);
+    ss << boost::format{"%-2s %11.6f %11.6f %11.6f\n"} % symbol % pos.x % pos.y % pos.z;
```

Comment:
Is the format space-sensitive so that it is important that the atom symbol be only 2 characters?

Reply:
I think usual parsers are space-insensitive; just intended to align columns.

Reply:
The problem with using `%-2s` is that it will, I believe truncate three-letter element names like `Uut`. I realize these aren't exactly common, but it's easy to not create the problem in the first place

Reply:
Three or more letters are truncated if the format specifier is `%-.2s`, while not truncated if `%-2s`. A three-letter element is just misaligned:

```cpp
#include <iostream>
#include <boost/format.hpp>
int main() {
  std::cout << boost::format{"%-2s %11.6f %11.6f %11.6f\n"} % "H" % 1. % 2. % 3.;
  std::cout << boost::format{"%-2s %11.6f %11.6f %11.6f\n"} % "Li" % -1.e3 % -2.e3 % -3.e3;
  std::cout << boost::format{"%-2s %11.6f %11.6f %11.6f\n"} % "Uut" % 1. % 2. % 3.;
  return 0;
}
```
gives
```
H     1.000000    2.000000    3.000000
Li -1000.000000 -2000.000000 -3000.000000
Uut    1.000000    2.000000    3.000000
```

Anyway, I increased the width to three.